crates/edr_provider/src/error.rs

```diff
@@ -399,10 +399,13 @@ fn revert_error(output: &Bytes) -> String {
             }
         }
         Err(decode_error) => match decode_error {
-            alloy_sol_types::Error::TypeCheckFail { .. } => {
+            alloy_sol_types::Error::TypeCheckFail { .. }
+            | alloy_sol_types::Error::UnknownSelector { .. } => {
                 format!("VM Exception while processing transaction: reverted with an unrecognized custom error (return data: 0x{})", hex::encode(output))
             }
-            _ => unreachable!("Since we are not validating, no other error can occur"),
+            _ => unreachable!(
+                "Since we are not validating, no other error can occur: {decode_error:?}"
+            ),
```

Comment:
@agostbiro you're more familiar with this code. Can we logically rule out that none of the other error variants can occur?

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4811/commits/3e86c0d9be3bcf6c1ba195134449e6c39f01e3b2